lib/ae_mdw/db/contract.ex

```diff
@@ -76,16 +76,35 @@ defmodule AeMdw.Db.Contract do
     :ok
   end
 
-  @spec aex9_presence_exists?(pubkey(), pubkey(), integer() | nil) :: boolean()
-  def aex9_presence_exists?(contract_pk, account_pk, txi \\ nil) do
-    txi_search_prev = (txi && txi + 1) || nil
+  @spec aex9_presence_exists?(pubkey(), pubkey(), integer()) :: boolean()
+  def aex9_presence_exists?(contract_pk, account_pk, txi) do
+    txi_search_prev = txi + 1
 
     case :mnesia.prev(Model.Aex9AccountPresence, {account_pk, txi_search_prev, contract_pk}) do
       {^account_pk, _txi, ^contract_pk} -> true
       _other_key -> false
     end
   end
 
+  @spec aex9_presence_exists?(pubkey(), pubkey()) :: boolean()
+  def aex9_presence_exists?(contract_pk, account_pk) do
+    0
+    |> Stream.unfold(fn
+      :found ->
+        nil
+
+      txi_search_next ->
+        case :mnesia.next(Model.Aex9AccountPresence, {account_pk, txi_search_next, contract_pk}) do
+          {^account_pk, _txi, ^contract_pk} -> {true, :found}
+          {^account_pk, next_txi, _contract_pk} -> {false, next_txi}
+          _not_found -> nil
+        end
+    end)
+    |> Enum.to_list()
+    |> List.last()
+    |> Kernel.||(false)
+  end
```

Comment:
I thought of a more brief version of it, taking advantage that streams are lazy, and not all values will be computed:
```elixir
0
|> Stream.unfold(fn
  txi_search_next ->
    case :mnesia.next(Model.Aex9AccountPresence, {account_pk, txi_search_next, contract_pk}) do
      {^account_pk, next_txi, contract_pk} -> {contract_pk, next_txi}
      _not_found -> nil
    end
end)
|> Enum.member?(contract_pk)
```

Reply:
The `{^account_pk, _txi, ^contract_pk} -> {true, :found}` is helping to stop on first `contract_pk` match. Without this check, it iterates through all presences of the account.

The index gonna be refactored though to avoid these potential longer iterations #459

Reply:
> The `{^account_pk, _txi, ^contract_pk} -> {true, :found}` is helping to stop on first `contract_pk` match. Without this check, it iterates through all presences of the account.
> 
> The index gonna be refactored though to avoid these potential longer iterations #459

That's not how streams work, the first match with `member?` will stop the stream from computing the rest

Reply:
Please notice that the size of the list is determined by `Stream.unfold` (it doesn't use a condition outside of the inlined function).

Reply:
The benefit and elegance of streams, is that it could be infinite and not necessarily compute in its entirety. Manually handling the end of the stream adds complexity (e.g. having to handle end values like `:found` and adding a lot more code into it), but I guess it's a matter of taste, sorry for bring it up!

Reply:
No problem, I like discussing readability too. Only in this case it's a behaviour of `Stream.unfold`: "The enumeration finishes if it returns nil."

Maybe a case for a new Enum or Stream function to pick the last based not on a list but on unfolded result like a `reduce` but receiving a seed like `unfold`.

Reply:
"The enumeration finishes if it returns `nil`" doesn't mean that it will continue executing until it returns `nil`, all streams are lazy and will only execute if the value is needed. Example:
```elixir
Stream.unfold(0, fn 
  5 -> nil
  n -> IO.inspect(n + 1); {n + 1, n + 1} 
end) |> Enum.member?(3)
# prints 1, 2, 3
# returns true
```

As you can see in this example, the values 4 and 5 weren't computed at all